src/openms/include/OpenMS/CONCEPT/ClassTest.h

```diff
@@ -275,7 +275,9 @@ namespace OpenMS
       {
         ++test_count;
         test_line = line;
-        this_test = bool(expression_1 == T1(expression_2));
+        // cast to underlying type to be able to compare enums
+        this_test = bool(static_cast<std::underlying_type_t<T1>>(expression_1) ==
```

Comment:
PreviouslyI used:
TEST_TRUE(func_that_returns_enum == the_enum)
because
TEST_EQUAL(func_that_returns_enum, the_enum) didn't work
let's see if this is sufficient...